clients/src/main/java/org/apache/kafka/clients/NodeApiVersions.java

```diff
@@ -123,21 +125,25 @@ public short latestUsableVersion(ApiKeys apiKey) {
      * Get the latest version supported by the broker within an allowed range of versions
      */
     public short latestUsableVersion(ApiKeys apiKey, short oldestAllowedVersion, short latestAllowedVersion) {
```

Comment:
We could probably simplify this so that it takes a single `ApiVersion` parameter?

By the way, the implementation above `latestUsableVersion(ApiKeys apiKey)` since it basically does an intersection of the latest supported version with itself. A little helper (say `latestSupportedOrThrow`) might simplify this.